src/VisualStudio/Core/Def/RoslynPackage.cs

```diff
@@ -152,16 +151,17 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
             // Ensure the options persisters are loaded since we have to fetch options from the shell
             LoadOptionPersistersAsync(this.ComponentModel, cancellationToken).Forget();
 
-            _workspace = this.ComponentModel.GetService<VisualStudioWorkspace>();
-
             await InitializeColorsAsync(cancellationToken).ConfigureAwait(true);
 
             // load some services that have to be loaded in UI thread
             LoadComponentsInUIContextOnceSolutionFullyLoadedAsync(cancellationToken).Forget();
 
-            _solutionEventMonitor = new SolutionEventMonitor(_workspace);
+            // We are at the VS layer, so we know we must be able to get the IGlobalOperationNotificationService here.
+            var globalNotificationService = this.ComponentModel.GetService<IGlobalOperationNotificationService>();
+            Contract.ThrowIfNull(globalNotificationService);
```

Comment:
There's an Assumes.Present() call you can also use out of the VS asserts library that includes the typename, which is otfen used at the VS layer for service things like this.

Reply:
Done.